# Assignment 3 - Question 4 : Hypothesis Testing Exercise Customer Order Form

# Problem Statement : TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


### We are going to conduct a Test of Independence using Chi-Square Test with Contingency Table.

### We need to check whether there is an impact of center on defective form %

In [2]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [5]:
data = pd.read_csv('C:/Users/LENOVO/Desktop/Assignment Data Science/3rd/Costomer_OrderForm.csv')

data.head(8)

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free
5   Error Free  Error Free  Error Free  Error Free
6   Error Free   Defective  Error Free  Error Free
7   Error Free  Error Free  Error Free  Error Free

# Declare Null and Alternate Hypothesis

### Null Hypothesis (Ho) : Independence of categorical variables (customer order forms defective % does not varies by centre) 

### Alternative hypothesis (Ha) : Dependence of categorical variables (customer order forms defective % varies by centre)

In [6]:
data.shape

(300, 4)

In [7]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [8]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [9]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [10]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [11]:
# Make a contingency table

obs = np.array([[271,267,269,280],[29,33,31,20]])

obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [14]:
# Chi-Square Contingency Independence Test

t_stat, p_value, dof, expect = stats.chi2_contingency(obs)

print("Test_Statistics = ",t_stat,"\n","p_value = ",p_value,"\n","dof = ",dof,"\n","Exp_array = ",expect)

Test_Statistics =  3.858960685820355 
 p_value =  0.2771020991233135 
 dof =  3 
 Exp_array =  [[271.75 271.75 271.75 271.75]
 [ 28.25  28.25  28.25  28.25]]


In [15]:
p_value

0.2771020991233135

In [16]:
alpha = 0.05 # Level of significance

if p_value <= alpha:
    print('We reject Null Hypothesis')
else:
    print('We fail to reject Null hypothesis')

We fail to reject Null hypothesis


# As (p_value = 0.2771) > (α = 0.05); Accept Null Hypthesis i.e. Independence of categorical variables Thus, customer order forms defective % does not varies by centre.